exporter/helpers.go

```
@@ -291,12 +294,15 @@ func getDriveEndpoint(url, host string, client *retryablehttp.Client) (oem.Gener
 	if err != nil {
 		return drive, err
 	}
-	defer resp.Body.Close()
+	defer common.EmptyAndCloseBody(resp)
 	if !(resp.StatusCode >= http.StatusOK && resp.StatusCode < http.StatusMultipleChoices) {
 		if resp.StatusCode == http.StatusNotFound {
 			for retryCount < 3 && resp.StatusCode == http.StatusNotFound {
```

Since err is not checked inside the loop, and common.DoRequest returns either a nil err or nil resp, this probably means that there is an ull deref here in case of error. (the bug was already here AFAIU)

For this one I am sure since there is more than one retry :-)

---

Error check inside the loop has been added. Please pull the latest changes.